paasta_tools/flink_tools.py

```diff
@@ -361,3 +409,443 @@ def get_flink_overview_from_paasta_api_client(
         service=service,
         instance=instance,
     )
+
+
+def get_flink_instance_details(
+    metadata: Mapping[str, Any],
+    flink_config: Optional[Any],
+    flink_instance_config: Any,
+    service: str,
+    soa_dir: str = DEFAULT_SOA_DIR,
+) -> FlinkInstanceDetails:
+    """Collect Flink instance metadata and configuration details.
+
+    :param metadata: Kubernetes metadata from flink status
+    :param flink_config: Flink configuration (None if not running)
+    :param flink_instance_config: Flink instance config from yelpsoa-configs
+    :param service: Service name
+    :param soa_dir: SOA directory path
+    :returns: Dict with instance details
+    """
+    from paasta_tools.monitoring_tools import get_runbook
+    from paasta_tools.monitoring_tools import get_team
```

Comment:
we don't really do local imports like these - these should be up at the top with the rest